### PR TITLE
[FLINK-27160] Add e2e tests for session job

### PR DESCRIPTION
**Pull request by Aitozi**
This PR is meant to add the e2e test for the session job. It mostly aligned to the flinkdep tests.

### CODE REVIEW

**Comment by Aitozi** [comment]
cc @wangyang0918 @gyfora

**Comment by Aitozi** [comment]
Besides, I found that the test can not completely run on the Mac now. Because the minikube ip can not be reached from the local machine. It's a bit confusing ~

**Comment by wangyang0918** [comment]
The result of `minikube ip` should be accessible from local machine. What's driver of minikube? I am using `hyperkit` and it works well.

**Comment by Aitozi** [comment]
> The result of `minikube ip` should be accessible from local machine. What's driver of minikube? I am using `hyperkit` and it works well.

I use the `Docker` as the driver. It seems due to this: https://github.com/kubernetes/minikube/issues/11193#issuecomment-826331511

File: e2e-tests/test_sessionjob_savepoint_upgrade.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    kubectl wait --for=condition=Available --timeout=${TIMEOUT}s deploy/${CLUSTER_ID} || exit 1
+    jm_pod_name=$(kubectl get pods --selector="app=${CLUSTER_ID},component=jobmanager" -o jsonpath='{..metadata.name}')
+
+    echo "Waiting for jobmanager pod ${jm_pod_name} ready."
+    kubectl wait --for=condition=Ready --timeout=${TIMEOUT}s pod/$jm_pod_name || exit 1
+
+    wait_for_logs $jm_pod_name "Rest endpoint listening at" ${TIMEOUT} || exit 1
+}
+
+function assert_available_slots() {
```
Comment (wangyang0918) [inline e2e-tests/test_sessionjob_savepoint_upgrade.sh:51]: Could be extracted to `util.sh` and deduplicated.

**Review by wangyang0918 — COMMENTED** [review]
I left some minor comments.

File: e2e-tests/test_last_state_upgrade.sh [hunk 2]
```diff
@@ -47,6 +48,7 @@ function wait_for_jobmanager_running() {
 function assert_available_slots() {
   expected=$1
   ip=$(minikube ip)
+  curl http://$ip/default/${CLUSTER_ID}/overview
```
[the remark this answers is not in the document]
Comment (Aitozi) [reply e2e-tests/test_last_state_upgrade.sh:51]: Fixed

**Comment by Aitozi** [comment]
I have addressed your comments, please take a look again @wangyang0918

**Review by wangyang0918 — APPROVED** [review]
+1 for merging.

**Pull request merged**